addons/procurement/procurement_view.xml

```diff
@@ -292,5 +292,22 @@
                 </form>
             </field>
         </record>
+
+        <record id="procurement_order_server_action" model="ir.actions.server">
+            <field name="name">Run Procurements</field>
+            <field name="model_id" ref="model_procurement_order"/>
+            <field name="state">code</field>
+            <field name="code">self.run(cr, uid, context.get('active_ids'), context=context)</field>
```

Comment:
can you check with active_model also ? like
if context.get('active_model') == 'procurement_order' and context.get('active_ids'):